### PR TITLE
Revert "fix: Disable search on workspaces page"

### PR DESCRIPTION
**Pull request by SUPERCILEX**
What is the motivation for cf34204fd965e5a69dc2e618a5d1c0365a8ddf45? Please don't break people's existing workflows.

### CODE REVIEW

**Comment by jmbuhr** [comment]
I, too, would love for this feature to remain. I understand the idea behind pushing the launcher over the gnome-shell search, but it is not quite there yet and I am not ready to let go of the ease of use of overview + search (see https://github.com/pop-os/shell/issues/995)

**Comment by ArthurVardevanyan** [comment]
I also would love this feature to remain. Could it get added as a toggle, defaulted to off?

**Comment by mmstick** [comment]
Search is on the applications page where it belongs

**Comment by DuttaAB-dev** [comment]
> Search is on the applications page where it belongs

This is not an answer to the problem of your bad ux in anyway. Atleast then make the pop launcher more useful and intuitive(file searching, scrollable, contact searching etc etc...)

**Comment by SUPERCILEX** [comment]
@mmstick please reconsider. Users are telling you they don't want this change... You're making pop os worse without a valid reason.

**Comment by mmstick** [comment]
Honestly, it's bad UX to have search in the workspaces view. That's not what the workspaces view is for. There's already an applications view with the same search functionality.

**Comment by jmbuhr** [comment]
If we ignore some of the hostility in this thread, there are valuable insights to be gained here. Please keep in mind, that the folks at pop are making this cool piece of software available for free. On the other hand, the beta testers are donating their time to test and deliver a better product. So everyone is on the same team.

Arguably, workspaces and applications are two separate things, though the launcher attempts to unify everything as well. The real question is now, whether starting to type in one context constitutes a sufficient intent to change the context. I would argue, that this does not violate the separation of showing workspaces and searching applications.

**Comment by DuttaAB-dev** [comment]
The issue is not just about removing the search from workspaces. The issue is also about reducing the capabilities of the gnome search and then trying to replace it with something much more inferior. Gnome search works as an all stop solution to what ever the user wants to search, let it be apps, files, contacts, settings, characters, apps from the pop shop, calculations, you name it! But the Pop Launcher, in my point of view, is still way less powerful and polished. Yes, the idea is great to have a spot on search feature that appears just on your current working view without interrupting what your current work, just like Mac OS Spotlight and Krunner from KDE. But, Pop launcher still has a long way to go and would be unwise to replace the gnome search with the launcher until most of the features/functionalities have been implemented here as well so as to make it as a viable option for most users, instead of just Forcing the half baked version down everyone's throats.

I hope I was clear in presenting my opinion, have been a user of the Pop!_OS for over a year, been loving it so far and thus jumped onto the 21.04 beta train as soon as it was announced. Been loving what the great folks have been doing this far but, please do consider user's opinions, instead of throwing them under the bus. Regards.

**Comment by mmstick** [comment]
GNOME Search hasn't gone away. It's still there.

The pop-shell launcher does search apps, files, etc. It has plugin support similar to ulauncher.

**Comment by DuttaAB-dev** [comment]
I know it is still there. But, please don't reduce its capabilities. And if you do, please do add those capabilities to Pop Launcher instead. 
References: #3, pop-os/gnome-control-center#164 and pop-os/gnome-control-center#165

**Comment by SUPERCILEX** [comment]
Has this decision been discussed with the rest of the team? I see @jackpot51 is also on the commit.

@jmbuhr made a great point: **starting to type should be considered enough intent to switch to the applications tab**.

This change is very frustrating since the Meta key used to be the one stop shop for managing open applications and starting new ones with a GUI. I don't want to use the launcher UI: being able to see my workspaces and type to start a new app was awesome.

Maybe a better question is *who wants this change?* If no one has complained, then this should be reverted plain and simple. Ideal UX doesn't matter if users don't like it.

**Comment by mmstick** [comment]
Yes, all changes being made with COSMIC have been discussed and planned out in advance. You can see the project issue board for the COSMIC project [here](https://github.com/orgs/pop-os/projects/15), where our efforts are coordinated.

Submitting a pull request to revert a commit that is a direct solution for a task on the board isn't going to be accepted outright. You're free to discuss the issue and voice concerns, but for now this is what we're experimenting with.

The general issue I've had with the search in workspaces is that it causes people to never look at the applications view at all, or even realize that there is an applications view. Converting to a semi-applications view on key press may conflict with future changes we'd want to make to the workspaces experience in Pop.

**Comment by SUPERCILEX** [comment]
> Converting to a semi-applications view on key press may conflict with future changes we'd want to make to the workspaces experience in Pop.

Does this mean adding setting to allow restoring previous behavior is also off the table? Otherwise would you be open to a PR that does that?

**Comment by mmstick** [comment]
I've added a discussions section https://github.com/pop-os/cosmic/discussions

Which would be more useful than having a discussion on a closed issue. I think a discussion beforehand would be better than a drive by PR, so you avoid going through the work of pushing a PR that may or may not be accepted.

**Comment by SUPERCILEX** [comment]
Alright, thanks.

**Comment by jackpot51** [comment]
Previously, I did have typing on workspaces go to applications with the search. Disabling this, I think, is supposed to remind users of the division between the two new overviews, Workspaces and Applications. @maria-komarova could you read through these concerns and see what you think?

**Comment by ArthurVardevanyan** [comment]
If we are still discussing here, here is my feedback.
For Reference, My Perspective is a Triple Monitor Setup.

### TLDR: 
Workspace allows me to see all my apps, what is on the app, which monitor it is on, and allows me to quickly switch to another application, move it to another screen, _or search and open a new app/window if I don't see it._

For me, personally, I ended up changing it back, and installing it as an extension, and will continue to do it that for future updates.

### Detailed:

**Application:**
I rarely use the Applications Screen. (Even before cosmic)
I have my most common apps pinned on the bottom bar, but I generally prefer to search for most apps, it _feels_ quicker.
It depends where my cursor and mouse hand is at.

**Launcher:**
Used couple times to Test, but Never actual use it.
I personally don't see the added value of the launcher
I can't see what is on any of the applications, and I don't know where any of the applications are.

**Workspace,** 
I can see what applications I have open, and which monitor they are on, even if they are nested or hidden. 
I can then quickly switch to another application, move an application to another monitor, or search and open a new app/window if I don't see it.

**Comment by mmstick** [comment]
You can't make an informed opinion on something if you haven't actually used it. The general goal of the launcher is to be similar to something like Ulauncher, or popularized runners on tiling window managers like Rofi. It is a universal search just like the search in GNOME Shell, but it also allows you to search opened windows to focus. There are already some useful plugins, such as a search prefix with `=` activating a calculator mode, or `d:` listing recent documents in search results. It's not perfect, but we're getting there.

**Comment by ArthurVardevanyan** [comment]
> You can't make an informed opinion on something if you haven't actually used it. 

Not sure about other people, but I'm not really partial to whether it is gnome search or launcher.

In that case, on keystroke, instead of 
Workspace -> Gnome Search

It would do.
Workspace -> Launcher

I'll have to play around and see if I can configure cosmic to work that way.

**Comment by jackpot51** [comment]
@ArthurVardevanyan that is interesting to try.

We are thinking over the comments from here, and will come back with some decisions maybe tomorrow.

**Comment by jmbuhr** [comment]
I arrived at the same conclusion (typing to open launcher) in my launcher-wishlist (https://github.com/pop-os/shell/issues/995), so I am excited to try this.

**Comment by Avendattor** [comment]
I have started new [discussion](https://github.com/pop-os/cosmic/discussions/95) for this search.
